### ISSUE TITLE
p8011 not found.

### ISSUE THREAD

**Issue by rays** [pimusicbox/pimusicbox#202]
After upgrading to 0.5.2 from the last 0.4 stable version I'm getting the following on boot (some typos may appear as I'm typing this by hand). This error repeats and never does give me a prompt.

```
[warn] Running /etc/init.d/networking restart is deprecated because it may not re-enable some interfaces ... (warning).
[....] Reconfiguring network interface...catL /sys/module/p80211/parameters/wlan-uext_write: No such file or directory
FATAL: Module p80211 not found.
Failed to load p80211.ku.
run-parts: /etc/network/if-pre-up.d/linux-wlan-ng-pre-up exited with return code 1
Error for wireless request "SEt Power Management" (8B2C) :
    SSSET failed on device wlan0 : Operation not permitted.
ioctl[S10CSIWAP]: Operation not permitted
ioctl[S10CSIWENCODEEXT] Invalid argument
ioctl[S10CSIWENCODEEXT] Invalid argument
done.
```

(repeats)

**Comment by woutervanwijk** [comment]
yes. It's from the wifi. Should be fixed, but I don't know if it harms anyone?

**Comment by rays** [comment]
this is on boot, so this message keeps repeating and pimusicbox services never start

**Comment by mach76** [comment]
Same problem for me on both 5.3 and the 5.4 beta on B+ hardware. Never able to get Edimax ew-7811Un wifi dongle working with Pi-Musicbox code. Dongle works out of the box with 3 other Pi project builds.

**Comment by hadess** [comment]
I'm guessing that the Pi-MusicBox kernel build is just missing the driver, right?

**Comment by kingosticks** [comment]
It's based on raspbian. If raspbian are missing it then we are missing it.

**Comment by anielk** [comment]
we've got the same message here, is there already a fix for it?

**Comment by outerrim** [comment]
Exactly same message for me on Raspberry A+ with latest image (0.6)

**Comment by TheBox193** [comment]
Ditto 
Raspberry B
MusicBox v0.6

**Comment by kingosticks** [comment]
If you are able to get a prompt and login, or if you can do so using normal raspbian, it would be helpful to know exactly what USB wifi dongles have the issue (output of `lsusb`, `lsmod` would also be nice).
*

**Comment by TheBox193** [comment]
It appears to simply been an issue of a mistyped wifi network name. The loop seems to indicate it failed to connect or find a network by with that ssid/password. Once corrected I was able to connect with no issue. It might be nice to catch and fail with an error depending how common of an issue this is for users.

**Comment by ThreeSixes** [comment]
Similar problem with multiple attempts to reinstall on Raspberry Pi B+. 
At boot it fails to load p8011.ko, fails on "Reconfiguring network interfaces...cat: /sys/module/parameters/wlan_wext_write: No such file or directory", then Wlan0 link not ready... [infinite error loop doing this]

**Comment by kingosticks** [comment]
Are you using a Edimax ew-7811Un wifi dongle? If not, what are you using?

**Comment by ThreeSixes** [comment]
I'm using the TP-LINK TL-W725N adapter.

**Comment by kingosticks** [comment]
I don't get many search hits for that card but I do get a lot of stuff regarding TL-WN725N. Are they the same thing? If so, it seems that it might be the 8188eu driver it wants which needs to be compiled from source. I *think* we include it in pimusicbox but it's possible the binary that's included isn't compatible or that you've updated your system to a non-compatible kernel.  Since I don't have this hardware, and I can't find a link to buy a TP-LINK TL-W725N, it's tricky to debug.

Does it work in Raspbian? Are you able to remove the dongle, boot musicbox and login (locally or ssh via ethernet connection), insert the dongle, and then provide output of the following commands. 
* `lsusb`
* `sudo lshw -C network`

**Comment by ThreeSixes** [comment]
Hey Nick, it looks like the specific USB adapter I had was defective. I
swapped it for one of the same model and it started working. Whew!

**Comment by kingosticks** [comment]
Oh right! That's great, thanks very much for updating the issue. These
wireless sticks can be a right pain.

**Comment by kingosticks** [comment]
Just to update this topic, since I now have an TL-WN725N adapter to test with and all I can find online suggests that TL-W725N and TL-WN725N are the same thing. They are two versions, v1 and v2, of this device. v1 (along with the earlier mentioned Edimax EW-7811Un dongle) requires driver module rtl8192cu which is available in the kernel (and may or may not be any good). v2 requires r8188eu which is distributed in firmware-realtek but the version of this in Wheezy (v0.36) is broken as it does not include the required firmware blob. A fully working r8188eu driver (including firmware blob) is available in jessie's v0.43 version of firmware-realtek. 

Finally, from what I can see, if the correct driver is not available, the kernel picks one and for some reason this is p80211. If you try to do anything with it (e.g. ìfup wlan0`) it fails with the originally posted error message.

**Comment by kingosticks** [comment]
Thanks to @DavisNT who [helped me see](https://github.com/pimusicbox/pimusicbox/issues/303) that there's now a backported package providing the r8188eu driver available in the Raspberry Pi Foundation APT repository. Anything using this driver (such as TL-WN725N) will work out the box.

That p80211 error message is in fact due to a buggy script that's included in linux-wlan-ng which tries to always load the p80211 driver. Any wireless problems you are having are totally unrelated to this message. Since we apparently don't need [this package](https://packages.debian.org/wheezy/linux-wlan-ng) and it's just confusing things, I am keen to remove it.
